test/validation/api/ipsec/ipsec.c

```diff
@@ -781,6 +781,30 @@ static int ipsec_send_out_one(const ipsec_test_part *part,
 	return num_out;
 }
 
+static void ipsec_pkt_proto_err_set(odp_packet_t pkt)
+{
+	odph_ipv4hdr_t *ip = odp_packet_l3_ptr(pkt, NULL);
+
+	/* Simulate proto error by corrupting protocol field */
+
+	if (ip->proto == ODPH_IPPROTO_ESP)
+		ip->proto = ODPH_IPPROTO_AH;
+	else
+		ip->proto = ODPH_IPPROTO_ESP;
```

Comment:
This may write outside the packet segment if there is a segment boundary before the protocol field. The copy-to-and-from-mem pattern would fix this, or alternatively, maybe you could just pass non-null second parameter to odp_packet_l3_ptr() and then assert that the contiguous length is sufficient.

Reply:
Missed this. copy-to-and-from-mem looks good. I'll make that change.